src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueCollateralAdaptor.sol

```diff
@@ -0,0 +1,230 @@
+        return 0;
+    }
+
+    /**
+     * @notice Returns the cellar's balance of the collateralAsset position.
+     * @param adaptorData the collateral asset deposited into Morpho Blue
+     * TODO - could use the periphery library `MorphoBalancesLib` to get the expected balance (w/ simulated interest) but for now we just query the getter. We may switch to using the periphery library.
+     * @return Cellar's balance of provided collateral to specified MB market.
+     * @dev normal static call, thus msg.sender for sommelier focus is calling cellar
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        Id id = abi.decode(adaptorData, (Id));
+        return _userCollateralBalance(id, msg.sender);
+    }
+
+    /**
+     * @notice Returns collateral asset.
+     * @return The collateral asset in ERC20 type.
+     */
+    function assetOf(bytes memory adaptorData) public view override returns (ERC20) {
+        Id id = abi.decode(adaptorData, (Id));
+        MarketParams memory market = morphoBlue.idToMarketParams(id);
+        return ERC20(market.collateralToken);
+    }
+
+    /**
+     * @notice This adaptor returns collateral, and not debt.
+     * @return Whether or not this position is a debt position
+     */
+    function isDebt() public pure override returns (bool) {
+        return false;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to add collateral to the respective cellar position on specified MB Market, enabling borrowing.
+     * @param _id encoded bytes32 MB id that represents the MB market for this position.
+     * @param _collateralToDeposit The amount of `collateralToken` to add to specified MB market position.
+     */
+    function addCollateral(Id _id, uint256 _collateralToDeposit) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+        ERC20 collateralToken = ERC20(market.collateralToken);
+
+        uint256 amountToDeposit = _maxAvailable(collateralToken, _collateralToDeposit);
+        collateralToken.safeApprove(address(morphoBlue), amountToDeposit);
+
+        _addCollateral(market, amountToDeposit);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(collateralToken, address(morphoBlue));
+    }
+
+    /**
+     * @notice Allows strategists to remove collateral from the respective cellar position on specified MB Market.
+     * @param _id encoded bytes32 MB id that represents the MB market for this position.
+     * @param _collateralAmount The amount of collateral to remove from specified MB market position.
+     */
+    function removeCollateral(Id _id, uint256 _collateralAmount) public {
+        _validateMBMarket(_id);
+        MarketParams memory market = morphoBlue.idToMarketParams(_id);
+
+        _accrueInterest(market);
```

Comment:
Move inside if statement, see supply adaptor comment

Reply:
Hmm, I actually think we don't need `_accrueInterest()` in this function call. Collateral is not affected by interest accrual, just the health factor calculation. Removing collateral also calls `accrueInterest()` in morphoBlue contracts so it's already done within the mutative part of the function call.